src/main/java/dk/cs/aau/huppaal/presentations/RunConfigurationEditorPresentation.java

```diff
@@ -0,0 +1,168 @@
+                return;
+            }
+            getCurrentlySelectedConfiguration().ifPresent(c -> {
+                c.executionDir = f.getAbsolutePath();
+                populatePropertyGridPane(c);
+            });
+        });
+        controller.propertyGridPane.addRow(rc++, new StackPane(new Label("execution directory")), new StackPane(execDirFieldButton));
+
+        // Add environment variables field
+        environmentVariablesField.textProperty().addListener((b,o,n) -> getCurrentlySelectedConfiguration().ifPresent(c -> c.environmentVariables = n));
+        controller.propertyGridPane.addRow(rc++, new StackPane(new Label("environment")), new StackPane(environmentVariablesField));
+    }
+
+    private void populatePropertyGridPane(RunConfiguration r) {
+        nameField.setText(r.name);
+        if(!Strings.isNullOrEmpty(r.program))
+            programFieldButton.setText(r.program);
+        else
+            programFieldButton.setText("File");
+        argumentsField.setText(String.join(" ", r.arguments));
+        if(!Strings.isNullOrEmpty(r.executionDir))
+            execDirFieldButton.setText(r.executionDir);
+        else
+            execDirFieldButton.setText("Folder");
+    }
+
+    private void initializeOkCancelApplyButtons() {
+        controller.cancelButton.setOnAction(e -> closeWindow());
+        controller.applyButton.setOnAction(e -> saveRunConfigurations());
+        controller.okButton.setOnAction(e -> {
+            saveRunConfigurations();
+            closeWindow();
+        });
+    }
+
+    private void initializeAddAndRemoveButtons() {
+        controller.addNewRunConfigurationButton.setOnMouseClicked(e ->
+                controller.savedConfigurationsList.getItems().add(
+                        new RunConfiguration("new config", "", "", HUPPAAL.projectDirectory.get(), "")));
+        controller.removeSelectedRunConfigurationButton.setOnMouseClicked(e ->
+                getCurrentlySelectedConfiguration().ifPresent(runConfiguration ->
+                        controller.savedConfigurationsList.getItems().remove(runConfiguration)));
+    }
+
+    private void initializeRunConfigurationSelectionChange() {
+        controller.savedConfigurationsList.setOnMouseClicked(e -> getCurrentlySelectedConfiguration().ifPresent(this::populatePropertyGridPane));
+    }
+
+    private Optional<RunConfiguration> getCurrentlySelectedConfiguration() {
+        var c = controller.savedConfigurationsList.getSelectionModel().getSelectedItem();
+        if(c == null)
+            return Optional.empty();
+        return Optional.of(c);
+    }
+
+    private void saveRunConfigurations() {
+        HUPPAAL.preferences.put(RunConfigurationPreferencesKeys.ConfigurationsList, gson.toJson(controller.savedConfigurationsList.getItems()));
+        HUPPAAL.showToast("Saved RunConfigurations");
+        if(onRunConfigsSaved != null)
+            onRunConfigsSaved.run();
+    }
```

Comment:
Consider setting the selected configuration to the newly saved one here

Reply:
Fixed in 69b536fbcdcf24112773004796a5e97e20718c47